Userland/Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -30,10 +31,56 @@
 
 namespace Web::DOM {
 
+class NodeDirectory {
+public:
+    NodeDirectory()
+        : m_nodes_by_id()
+    {
+    }
+
+    u64 create_node_id(Node* node)
```

Comment:
Seems like a decent usecase for `AK::IDAllocator` :^)

Reply:
Oooh, that's very handy! Thanks!